transport/tchannel/channel_transport.go

```diff
@@ -82,11 +82,12 @@ func (options transportOptions) newChannelTransport() *ChannelTransport {
 		logger = zap.NewNop()
 	}
 	return &ChannelTransport{
-		once:   lifecycle.NewOnce(),
-		ch:     options.ch,
-		addr:   options.addr,
-		tracer: options.tracer,
-		logger: logger,
+		once:      lifecycle.NewOnce(),
+		ch:        options.ch,
+		addr:      options.addr,
+		tracer:    options.tracer,
+		logger:    logger,
+		rawHeader: options.rawHeader,
```

Comment:
rawHeader is not particularly descriptive. The option directs the transport to "use the exact case for header keys as specified in with-header call options, specifically for the case of forwarding headers unmodified."

Ideas:
- `caseInsensitiveHeaders`
- `preserveHeaderCase`
- `preserveHeaders`
- `preserveExactHeaders`
- `forwardExactHeaderKeys`

Thinking aloud: we don’t even really need to preserve case for all uses of With() header options, but only for the case of forwarding. Perhaps instead of enabling With() call-sites to preserve the case in code, we can add a copy constructor for headers that has the ability to preserve case. This would reduce the scope of the change to forwarding middleware, instead of opening up the possibility of originating exact case headers?